src/main/java/com/example/boot/service/RoleService.java

```diff
@@ -5,5 +5,5 @@
 public interface RoleService {
     Role save(Role role);
 
-    Role getByName(String name);
+    Role getByRoleName(Role.RoleName roleName);
```

Comment:
you shouldn't change the method in service - we should be able to pass String on the service level - and here it should be transformed into RoleName type